lib/bandit/headers.ex

```diff
@@ -97,6 +97,22 @@ defmodule Bandit.Headers do
       else: headers
   end
 
+  # Respect the content-length header (if set) for valid HEAD responses
+  # so that the handler can avoid rendering the body to specify its length.
+  @spec override_content_length?(
+          headers :: Plug.Conn.headers(),
+          status :: Plug.Conn.int_status(),
+          method :: Plug.Conn.method()
+        ) ::
+          boolean()
+
+  def override_content_length?(headers, status, "HEAD") do
+    value = get_header(headers, "content-length")
+    is_nil(value) or not add_content_length?(status)
```

Comment:
Upon further reflection, I actually believe that the server should only override the content-length of HEAD responses if the status code requires it to be omitted.

Specifically, the language in [RFC9110 §8.6](https://www.rfc-editor.org/rfc/rfc9110.html#name-content-length):
> A server MAY send a Content-Length header field in a response to a HEAD request; a server MUST NOT send Content-Length in such a response unless its field value equals the decimal number of octets that would have been sent in the content of a response if the same request had used the GET method.

* the server has no way of knowing what the length of the GET would have been unless it's provided as the body
* [§9.3.2](https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.2) is clear that HEAD requests should not generate and discard the content in order to populate response headers
* HEAD responses will never send the body, so it doesn't make sense to set the body to anything other than `""`
* the server is not required to set a Content-Length header for HEAD responses

When the application does not specify a Content-Length for HEAD responses, it would not be correct for the server to impose the response's Content-Length based on the size of the response body.